### ISSUE TITLE
Allow `undefined` and `null` in direct comparison 

### ISSUE THREAD

**Issue by tarikjabiri** [microsoft/TypeScript#50414]
# Allow `undefined` and `null` in direct comparison with `>`, `<`, `>=`, `<=`, `==` and `===` operators.

Look at code snippet:
```ts
interface Options {
	constantWidth?: number
}

function test(options: Options) {
	if(options.constantWidth && options.constantWidth > 0) {
		console.log('Defined and greater than 0')
	} else {
		console.log('Undefined or less than 0')
	}
}

function testShort(options: Options) {
	// It is possible to allow this comparison ?
	if(options.constantWidth > 0) { // TS: Object is possibly 'undefined'.
		console.log('Defined and greater than 0')
	} else {
		console.log('Undefined or less than 0')
	}
}

test({}) // Undefined or less than 0
testShort({}) // Undefined or less than 0

test({constantWidth: 5}) // Defined and greater than 0
testShort({constantWidth: 5}) // Defined and greater than 0

// But typescript is complaining about the second function which is normal.
```
I suggest:
- In any comparison  `undefined` and `null` always return false so why not allow this type of comparison.
So we ends up with writing one condition.

Regards

**Comment by MartinJohns** [comment]
You forgot to fill out the issue template for feature requests.

You can just use the `??` operator to be explicit and type-safe: `options.constantWidth ?? 0 > 0`

Duplicate of #17801.

**Comment by tarikjabiri** [comment]
Thank you for the quick response.
`options.constantWidth ?? 0 > 0` this is great 👍
